### PR TITLE
DOC-865: Rework and reorder changelog entries to group similar items

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: DOC-852

Description of Changes:
* Fixed some changelogs being in the incorrect groups
* Reordered changelogs to keep similar items together
* Applied feedback from Tyler and Millie on changelog wording

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -16,78 +15,80 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-- Added new `FormatApply` and `FormatRemove` events #TINY-7713
 
 ### Improved
+- Sticky toolbars can now be offset from the top of the page using the new `toolbar_sticky_offset` setting #TINY-7337
+- Fancy menu items now accept an `initData` property to allow custom initialization data #TINY-7480
 - Improved the load time of the `fullpage` plugin by using the existing editor schema rather than creating a new one #TINY-6504
-- Improved the performance when rendering UI components #TINY-7572
-- When scrolling, the context toolbar will stick to where it was previously for large elements, such as tables #TINY-7545
+- Improved the performance when UI components are rendered #TINY-7572
+- When scrolling, the context toolbar will remain where it was previously for large elements, such as tables #TINY-7545
```
Comment (metricjs) [inline modules/tinymce/CHANGELOG.md:35]:
```suggestion
- When scrolling, context toolbars will not reposition unneccessarily for large elements, instead remaining where they are if the position is valid
```
???? not exactly what I was thinking earlier, but along the same lines

File: modules/tinymce/CHANGELOG.md [hunk 2]
```diff
@@ -16,78 +15,80 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ### Fixed
-- The `dir` attribute was being incorrectly applied to list items #TINY-4589
-- `editor.fire` was incorrectly mutating the original `args` provided #TINY-3254
+- The `editor.fire` API was incorrectly mutating the original `args` provided #TINY-3254
+- Unbinding an event handler did not take effect immediately while the event was firing #TINY-7436
+- Binding an event handler incorrectly took effect immediately while the event was firing #TINY-7436
+- Unbinding a native event handler inside the `remove` event caused an exception that blocked editor removal #TINY-7730
 - The `SetContent` event contained the incorrect `content` when using the `editor.selection.setContent()` API #TINY-3254
 - The editor content could be edited after calling `setProgressState(true)` in iframe mode #TINY-7373
 - Tabbing out of the editor after calling `setProgressState(true)` was inconsistent in iframe mode #TINY-7373
```
Comment (metricjs) [inline modules/tinymce/CHANGELOG.md:59]:
```suggestion
- Tabbing out of the editor after calling `setProgressState(true)` behaved inconsistently in iframe mode #TINY-7373
```
maybe?

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -16,78 +15,80 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Fixed table corruption when deleting a `contenteditable="false"` cell #TINY-7891
+- The `dir` attribute was being incorrectly applied to list items #TINY-4589
+- Applying selector formats would sometimes not apply the format correctly to elements in a list #TINY-7393
+- For formats that specify an attribute or style that should be removed, the formatter `match` API incorrectly returned `false` #TINY-6149
+- The type signature on the `formatter.matchNode` API had the wrong return type (was `boolean` but should have been `Formatter | undefined`) #TINY-6149
+- The `formatter.formatChanged` API would ignore the `similar` parameter if another callback had already been registered for the same format #TINY-7713
+- The `formatter.formatChanged` API would sometimes not run the callback the first time the format was removed #TINY-7713
 - Base64 encoded images with spaces or line breaks in the data URI were not displayed correctly. Patch contributed by RoboBurned
 
 ### Deprecated
```
Comment (metricjs) [inline modules/tinymce/CHANGELOG.md:94]: for the line below:

Hm. spellchecker was actually deprecated in 5.4 I think, though we must have only mentioned it in docs + blogposts. This work was where we added a console log. But still, not sure this changelog entry is technically correct based on that?

**Review by metricjs — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 1 above)
Comment (JamesToohey) [reply modules/tinymce/CHANGELOG.md:35]:
```suggestion
- The context toolbar will no longer unnecessarily reposition to the top of large elements when scrolling #TINY-7545
```

**Review by ashfordneil — APPROVED** [review]

**Pull request merged**